src/sidebar/services/api.js

```diff
@@ -152,30 +90,42 @@ function createAPICall($http, $q, links, route, tokenGetter) {
           headers.Authorization = 'Bearer ' + token;
         }
 
-        const req = {
-          data: data ? stripInternalProperties(data) : null,
-          headers: headers,
+        const { url, params: queryParams } = replaceURLParams(
+          descriptor.url,
+          params
+        );
+        const apiUrl = new URL(url);
+        apiUrl.search = queryString.stringify(queryParams);
+
+        return fetch(apiUrl.toString(), {
+          body: data ? JSON.stringify(stripInternalProperties(data)) : null,
+          headers,
           method: descriptor.method,
-          params: url.params,
-          paramSerializer: serializeParams,
-          url: url.url,
-        };
-        return $http(req);
+        });
       })
-      .then(function(response) {
-        if (options.includeMetadata) {
-          return { data: response.data, token: accessToken };
+      .then(response => {
+        let data;
+
+        const status = response.status;
+        if (status >= 200 && status !== 204 && status < 500) {
```

Comment:
Here I'm assuming that if the response is >= 200 and < 500 that, except for 204, the content type will be JSON. Does that sound right @lyzadanger?

Reply:
For any service listed in the service descriptor set, this is true (technically, 204s also set their Content-Type to `application/json` but their bodies are empty. The `api_config` decorator makes this happen. There are some one-off views in the `auth` [(OAuth) API views](https://github.com/hypothesis/h/blob/95505e4e000ecb9a5c6025e433d872bdd080b1e7/h/views/api/auth.py) you may want to take a peek at if you have concerns.

Reply:
Oh, wait! 500-series errors are also JSON (with a body), see [these API error views](https://github.com/hypothesis/h/blob/95505e4e000ecb9a5c6025e433d872bdd080b1e7/h/views/api/errors.py#L57)

Reply:
That view says it is for "expected/deliberately thrown" errors. That implies 400-series errors to me, unless I missed something? In the case of 500 errors, that could come from multiple places depending on which part of the backend stack threw an error (Cloudflare, AWS, the Elastic Beanstalk container, nginx, the Python app) so I don't think we can rely on the response type in that case.

Reply:
(I had an offline conversation with Lyza, she clarified that 500 errors _may_ be in our standard JSON format _if_ they come from h and not some other part of the stack).